lib/processor.spec.js

```diff
@@ -0,0 +1,203 @@
+            }
+          }
+        }
+      }
+    }
+  )
+
+  processor.processFile({
+    contentful: {}
+  }, {
+    access_token: 'global-bar',
+    host: 'global-baz',
+    space_id: 'global-foo'
+  }).then(_ => t.end())
+})
+
+test.cb('processor.processFile - resolves correctly for a single entry', t => {
+  const entryToBeReturned = {
+    sys: {
+      id: 'bazinga',
+      contentType: {
+        sys: {
+          id: 'boing'
+        }
+      }
+    },
+    fields: {
+      name: 'John Doe'
+    }
+  }
+
+  const processor = proxyquire(
+    './processor',
+    {
+      contentful: {
+        createClient: function () {
+          return {
+            entries: function () {
+              return new Promise(resolve => resolve([ entryToBeReturned ]))
+            }
+          }
+        }
+      }
+    }
+  )
+
+  processor.processFile({
+    contentful: {
+      entry_id: '123456789'
+    },
+    _fileName: 'awesome-file.html'
+  }, {
+    access_token: 'global-bar',
+    host: 'global-baz',
+    space_id: 'global-foo'
+  }).then(fileMap => {
+    t.is(typeof fileMap['awesome-file.html'], 'object')
+    t.is(fileMap['awesome-file.html'].data, entryToBeReturned)
+
+    t.end()
+  }).catch(console.log)
+})
+
+test.cb('processor.processFile - resolves correctly for multiple entries', t => {
+  const entriesToBeReturned = [
+    {
+      sys: {
+        id: 'bazinga',
+        contentType: {
+          sys: {
+            id: 'boing'
+          }
+        }
+      },
+      fields: {
+        name: 'John Doe'
+      }
+    },
+    {
+      sys: {
+        id: 'badabum',
+        contentType: {
+          sys: {
+            id: 'kazum'
+          }
+        }
+      },
+      fields: {
+        name: 'Jane Doe'
+      }
+    }
+  ]
+
+  const processor = proxyquire(
+    './processor',
+    {
+      contentful: {
+        createClient: function () {
+          return {
+            entries: function () {
+              return new Promise(resolve => resolve(entriesToBeReturned))
+            }
+          }
+        }
+      }
+    }
+  )
+
+  /* eslint-disable no-template-curly-in-string */
+  processor.processFile({
+    contentful: {
+      content_type: 'article',
+      entry_filename_pattern: 'article/${ fields.name }',
+      entry_template: 'template.html'
+
+    },
+    _fileName: 'awesome-file.html'
+  }, {
+    access_token: 'global-bar',
+    host: 'global-baz',
+    space_id: 'global-foo'
+  }).then(fileMap => {
+    t.is(typeof fileMap['article/john-doe.html'], 'object')
+    t.is(fileMap['article/john-doe.html'].contents, '')
+    t.is(fileMap['article/john-doe.html'].data, entriesToBeReturned[0])
+    t.is(fileMap['article/john-doe.html'].id, entriesToBeReturned[0].sys.id)
+    t.is(fileMap['article/john-doe.html'].contentType, 'article')
+    t.is(fileMap['article/john-doe.html'].layout, 'template.html')
+    t.is(fileMap['article/john-doe.html']._fileName, 'article/john-doe.html')
```

Comment:
Don't really know whats best practice in tap, but I generally prefer a deep-eq over asserting each property as deep-eq ensures that really all properties are covered.